plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java

```diff
@@ -4079,6 +4079,41 @@ else if (material.hasModernData() && material.getModernData() instanceof org.buk
             return new ElementTag(BukkitScriptEvent.inCheckInternal(attribute.context, "is_in tag", object, attribute.getParam(), "is_in tag", "is_in tag"));
         });
 
+        // <--[tag]
+        // @attribute <LocationTag.last_interacted_slot>
+        // @returns ElementTag(Number)
+        // @description
+        // Returns the last interacted slot of a Chiseled Bookshelf inventory.
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "last_interacted_slot", (attribute, object) -> {
+            BlockState state = object.getBlockStateForTag(attribute);
+            if (!(state instanceof ChiseledBookshelf)) {
+                return null;
+            }
+            return new ElementTag(((ChiseledBookshelf) object.getBlockState()).getLastInteractedSlot() + 1);
+        });
+
+        // <--[tag]
+        // @attribute <LocationTag.slot[<vector>]>
+        // @returns ElementTag(Number)
+        // @description
+        // Returns the appropriate slot based on a vector relative to this block.
+        // It will return -1 if the given vector is not within the bounds of any slot.
+        // @Example
+        // # Obtain and narrate the slot that the player right-clicked.
+        // on player right clicks chiseled_bookshelf:
+        // - define slot <player.eye_location.ray_trace.sub[<context.location>]>
+        // - narrate <context.location.slot[<[slot]>]>
+        // See also <@link tag EntityTag.bookshelf_slot>
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "slot", (attribute, object) -> {
```

Comment:
Use the `registerTag` method that takes a param type instead of manually using `Attribute#paramAsType`